api/v1alpha1/eso_types.go

```diff
@@ -0,0 +1,114 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&ExternalSecretsOperator{}, &ExternalSecretsOperatorList{})
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//+kubebuilder:object:root=true
+
+// ExternalSecretsOperatorList is a list of ExternalSecretsOperator objects.
+type ExternalSecretsOperatorList struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is the standard list's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ListMeta `json:"metadata"`
+	Items           []ExternalSecretsOperator `json:"items"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+
+// ExternalSecretsOperator describes configuration and information about the managed external-secrets
+// deployment. The name must be `cluster` to make ExternalSecretsOperator a singleton that is, to
+// allow only one instance of ExternalSecretsOperator per cluster.
+//
+// ExternalSecretsOperator is mainly for configuring the global options and enabling the features, which
+// serves as a common/centralized config for managing multiple controllers of the operator. The object
+// will be created during the operator installation.
+//
+// +kubebuilder:validation:XValidation:rule="self.metadata.name == 'cluster'",message="ExternalSecretsOperator is a singleton, .metadata.name must be 'cluster'"
+// +operator-sdk:csv:customresourcedefinitions:displayName="ExternalSecretsOperator"
+type ExternalSecretsOperator struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the ExternalSecretsOperator.
+	// +kubebuilder:validation:Required
+	Spec ExternalSecretsOperatorSpec `json:"spec,omitempty"`
+
+	// status is the most recently observed status of the ExternalSecretsOperator.
+	Status ExternalSecretsOperatorStatus `json:"status,omitempty"`
+}
+
+// ExternalSecretsOperatorSpec is the specification of the desired behavior of the ExternalSecretsOperator.
+type ExternalSecretsOperatorSpec struct {
+	// globalConfig is for configuring the external-secrets-operator behavior.
```

Comment:
```suggestion
	// globalConfig is for configuring the external-secrets-operator behaviour.
```

Reply:
This requires to be updated at few more places in same file.